Objects/object.c

```diff
@@ -2519,6 +2519,34 @@ _PyObject_SetDeferredRefcount(PyObject *op)
 #endif
 }
 
+int
+PyUnstable_Object_EnableDeferredRefcount(PyObject *op)
+{
+#ifdef Py_GIL_DISABLED
+    if (!PyType_IS_GC(Py_TYPE(op))) {
+        // Deferred reference counting doesn't work
+        // on untracked types.
+        return 0;
+    }
+
+    if (_PyObject_HasDeferredRefcount(op)) {
+        // Nothing to do
+        return 0;
+    }
+
+    uint8_t bits = _Py_atomic_load_uint8(&op->ob_gc_bits);
+    if (_Py_atomic_compare_exchange_uint8(&op->ob_gc_bits, &bits, bits | _PyGC_BITS_DEFERRED) == 0)
+    {
+        // Someone beat us to it!
+        return 0;
+    }
```

Comment:
This section, from the `_PyObject_HasDeferredRefcount` to the compare-exchange, should be executed atomically and it's not currently as written. It's possible for `_PyObject_HasDeferredRefcount` to be `false`, but then have it change by the time the `ob_gc_bits` field is loaded.

Reply:
Good catch, (I think) it should be fixed now.